e2e/pages/vm-sidebar.po.ts

```diff
@@ -1,11 +1,125 @@
+  group = `e2e_group_${this.generateID()}`;
+  aff = `aff_${this.generateID()}`;
+  EC = browser.ExpectedConditions;
+
+  clickColorChange() {
+    element(by.css('.color-preview-container')).click();
+    const color = element
+      .all(by.css('.color.ng-star-inserted'))
+      .first()
+      .click();
+    return element(by.css('div.color-preview.ng-star-inserted')).getAttribute('style');
+  }
+
+  setDescription(text) {
+    element(by.tagName('cs-inline-edit')).click();
+    element(by.css('cs-description mat-icon.mdi-pencil.mat-icon.mdi'))
+      .isPresent()
+      .then(() => {
+        element(by.css('cs-description mat-icon.mdi-pencil.mat-icon.mdi')).click();
+      });
+    element(by.tagName('textarea'))
+      .clear()
+      .sendKeys(text);
+    element(by.css('cs-description button[type=submit]')).click();
+    expect(element(by.css('cs-description span.text-content')).getText()).toEqual(text);
+  }
+
   getVMName() {
     return element(by.tagName('h4')).getText();
   }
 
+  clickEditGroup() {
+    const edit = element(by.css('cs-instance-group mat-icon.mdi-pencil.mat-icon.mdi'));
+    browser.wait(this.EC.elementToBeClickable(edit), 2000, 'No edit Group button').then(() => {
+      edit.click();
+      this.waitDialogModal();
+    });
+  }
+
+  clickAddAffGroup() {
+    const edit = element(by.css('cs-affinity-group mat-icon.mdi-plus.mat-icon.mdi'));
+    browser
+      .wait(this.EC.elementToBeClickable(edit), 2000, 'No add Affinity group button')
+      .then(() => {
+        edit.click();
+        this.waitDialogModal();
+      });
+  }
+
+  setNewAffGroup(name) {
+    element(by.css('input[formControlName=name]')).sendKeys(name);
+    element(by.css('.add-rule-button.mat-icon-button')).click();
+    browser.wait(
+      this.EC.textToBePresentInElement(
+        element(by.cssContainingText('.ng-star-inserted', name)),
+        name,
+      ),
+      5000,
+      'New affinity group was not added',
+    );
+    element(by.xpath(`//span[text()="${name}"]/ancestor::tr`)).click();
+    element
+      .all(by.css('.mat-button.mat-primary'))
+      .last()
+      .click();
+  }
+
+  setNewGroupOption(group) {
+    element(by.cssContainingText('.mat-radio-label-content', 'Create a new group')).click();
+    element(by.css('cs-create-update-delete-dialog input[name=textValue]'))
+      .clear()
+      .sendKeys(group);
+  }
+
+  isEnabledSubmitGroupButton() {
+    return element(by.css('mat-dialog-container button[type=submit]')).isEnabled();
+  }
+
+  clickSubmitGroupButton() {
+    element(by.css('mat-dialog-container button[type=submit]')).click();
+  }
+
+  clickCancelGroupButton() {
+    element(by.css('mat-dialog-container button[type=button]')).click();
+  }
+
+  setExistingGroupOption() {
+    element(by.name('selectValue')).click();
+    const group = element(by.css('mat-option[aria-selected=false] span.mat-option-text'));
+    const groupText = group.getText();
+    group.click();
+    return groupText;
```

Comment:
Why are there 4 lines of code with two constants? Here you can keep within 2 lines of code without constants.

Reply:
agreed with the reviewer: decided not to make changes